plugins/modules/get_certificate.py

```diff
@@ -297,7 +303,10 @@ def main():
                 ctx.check_hostname = False
                 ctx.verify_mode = CERT_NONE
 
-            cert = ctx.wrap_socket(sock, server_hostname=host).getpeercert(True)
+            if not server_name:
+                server_name = host
+                         
+            cert = ctx.wrap_socket(sock, server_hostname=server_name).getpeercert(True)
```

Comment:
How about removing the `if` above, and using the following instead:
```suggestion
            cert = ctx.wrap_socket(sock, server_hostname=server_name or host).getpeercert(True)
```